lisa/operating_system.py

```
@@ -220,6 +244,51 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
+    @property
+    def release_version(self) -> Version:
+        release_version = self._get_os_version().release
+        if Version.isvalid(release_version):
+            return Version.parse(release_version)
+
+        version, rest = self._coerce_version(release_version)
+        if not version:
+            raise LisaException(
+                f"The OS version release {rest} is not in a valid format"
+            )
+
+        return version
+
+    def _coerce_version(self, version: str) -> Tuple[Optional[Version], Optional[str]]:
+        """
+        Convert an incomplete version string into a semver-compatible Version
+        object
+
+        source -
+        https://python-semver.readthedocs.io/en/latest/usage.html#dealing-with-invalid-versions
```

This pattern doesn't help a lot. We can just parse the version, and let the exception raise out. The code will be more simple. If we found some instance are not supported, then customize it.